File: src/test/java/com/philhanna/flashcards/domain/TestBasicCard.java
package com.philhanna.flashcards.domain;

import static org.junit.jupiter.api.Assertions.assertEquals;

import org.junit.jupiter.api.Test;

import com.philhanna.flashcards.domain.BaseTest;
import com.philhanna.flashcards.domain.Card;

/**
 * Unit tests for BasicCard
 */
public class TestBasicCard extends BaseTest {

   // ==================================================================
   // Unit tests
   // ==================================================================

   @Test
   void toggleSwitchesQuestionAndAnswer() {
      Card card = new BasicCard("111", "222");
      card.toggle();
      assertEquals("222", card.getQuestion(), "Question is unexpected");
      assertEquals("111", card.getAnswer(), "Answer is unexpected");
   }

}

File: tests/domain/test_basic_card.py
# tests.domain.test_basic_card

import pytest

from flashcards.domain.card import BasicCard


def test_question_property():
    card = BasicCard("What is 2+2?", "4")
    assert card.question == "What is 2+2?"


def test_answer_property():
    card = BasicCard("What is 2+2?", "4")
    assert card.answer == "4"


def test_question_and_answer_are_independent():
    card = BasicCard("Q", "A")
    assert card.question != card.answer
